node/sc/vt_recovery_test.go

```diff
@@ -164,6 +163,62 @@ func TestBasicKLAYTransferRecovery(t *testing.T) {
 	assert.Equal(t, nil, vtr.Recover()) // nothing to recover
 }
 
+// TestKLAYTransferLongRangeRecovery tests a long block range recovery.
+func TestKLAYTransferLongRangeRecovery(t *testing.T) {
+	tempDir := os.TempDir() + "sc"
+	os.MkdirAll(tempDir, os.ModePerm)
+	oldMaxPendingTxs := maxPendingTxs
+	maxPendingTxs = 2
+	defer func() {
+		maxPendingTxs = oldMaxPendingTxs
+		if err := os.RemoveAll(tempDir); err != nil {
+			t.Fatalf("fail to delete file %v", err)
+		}
+	}()
+
+	// 1. Init dummy chain and do some value transfers.
+	info := prepare(t, func(info *testInfo) {
+		for i := 0; i < testTxCount; i++ {
+			ops[KLAY].request(info, info.localInfo)
+			for i := uint64(0); i < filterLogsStride; i++ {
+				info.sim.Commit()
+			}
+		}
+	})
+	vtr := NewValueTransferRecovery(&SCConfig{VTRecovery: true}, info.localInfo, info.remoteInfo)
+
+	// 2. Update recovery hint.
+	err := vtr.updateRecoveryHint()
+	if err != nil {
+		t.Fatal("fail to update value transfer hint")
+	}
+	t.Log("value transfer hint", vtr.child2parentHint)
+	assert.Equal(t, uint64(testTxCount), vtr.child2parentHint.requestNonce)
+	assert.Equal(t, uint64(testTxCount-testPendingCount), vtr.child2parentHint.handleNonce)
+
+	// 3. Request events by using the hint.
+	err = vtr.retrievePendingEvents()
+	if err != nil {
+		t.Fatal("fail to retrieve pending events from the bridge contract")
+	}
+
```

Comment:
After 1st `retrievePendingEvents`, how about checking the count of event recovered?

Reply:
Fixed.